src/main/java/org/mockito/ArgumentCaptor.java

```diff
@@ -100,12 +84,41 @@ private ArgumentCaptor(Class<? extends T> clazz) {
      * @return null or default values
      */
     public T capture() {
-        Mockito.argThat(capturingMatcher);
+        return captureIf(Any.ANY);
```

Comment:
With this we will add to much restriction. Maybe there are users that rely on the fact on the current implementation there is no type check when capturing arguments, so I would leave it like this for backward compatibility.